Cognite.Simulator.Utils/ModelLibraryBase.cs

```diff
@@ -599,19 +581,23 @@ private async Task<bool> DownloadFileAsync(T modelState)
 
             try
             {
-                var response = await _cdfFiles
+                var fileRes = await _cdfFiles
+                    .RetrieveAsync([fileId])
+                    .ConfigureAwait(false);
+                var file = fileRes.FirstOrDefault();
+
+                var downloadUriRes = await _cdfFiles
                     .DownloadAsync(new[] { fileId })
                     .ConfigureAwait(false);
-                if (response.Any() && response.First().DownloadUrl != null)
-                {
-                    var uri = response.First().DownloadUrl;
 
-                    string filename;
+                if (file != null && downloadUriRes.Any() && downloadUriRes.First().DownloadUrl != null)
+                {
+                    var uri = downloadUriRes.First().DownloadUrl;
 
-                    var modelFolder = _modelFolder;
-                    var storageFolder = Path.Combine(modelFolder, $"{modelState.CdfId}");
+                    var fileExtension = file.GetExtension();
+                    var storageFolder = Path.Combine(_modelFolder, $"{file.Id}");
+                    var filename = Path.Combine(storageFolder, $"{file.Id}.{fileExtension}");
                     CreateDirectoryIfNotExists(storageFolder);
-                    filename = Path.Combine(storageFolder, $"{modelState.CdfId}.{modelState.FileExtension}");
                     modelState.IsInDirectory = true;
```

Comment:
not something with this PR, but this should ideally be moved to inside the`if downlaoded` part so its only set when a model is actually in the directory.

Reply:
it is always in the directory now, check `storageFolder` and `CreateDirectoryIfNotExists()` above.
isInDirectory says whether it's a file or a directory, but from some time ago we only use directories.

Reply:
actually can/will change this in the next PR, as there will be changes related to it